content/blog/2019-11-mio-v0.7-alpha.md

```diff
@@ -0,0 +1,201 @@
+`kqueue` or `epoll_event` structures) would be converted into Mio's `Event`
+type. In version 0.7 the `Event` was changed to simply be a type wrapper around
+`kqueue` or `epoll_event` and provides convenience methods to get the event
+readiness indicators such as readable or writable. Similar changes have been
+made throughout the crate, for example `Poll` is now just a file descriptor on
+Unix.
+
+Handling of events in the `Events` has also changed. In version 0.6.10 index
+access was already deprecated and replaced with an iteration API. The index
+access was completely removed in version 0.7 and the iteration API was changed
+to return a reference to `Event` rather then making a copy, but since all
+methods on `Event` only need a reference this shouldn't be a problem.
+
+
+### Removal of the user space queue & deprecated types
+
+Mio version 0.6 had a user space queue, available through the `SetReadiness` and
+`Registration` types, but this has been removed in version 0.7 as it was deemed
+to be outside of the scope for Mio. Users who need a user space queue can use
+one of the queue types found in the [crossbeam] crate.
+
+One use case the user space queue support was waking a thread polling (i.e.
+calling `Poll::poll`) from another thread. To support this use case a new
+`Waker` type was introduced to do exactly this job, calling `Waker::wake` will
+wake the `Poll` it is associated with.
+
+[crossbeam]: https://crates.io/crates/crossbeam
+
+
+### Registering & I/O resource usages changes
+
+By far the biggest user facing change of version 0.7 is the way Mio registers
+I/O sources. In version 0.6 resource would be registered using a `Token`,
+`Ready` interests and `PollOpt`ions. For example the following code would register
+`socket` with readable interests and edge triggers.
+
+```rust
+poll.register(&socket, Token(0), Ready::readable(), PollOpt::edge())?;
+```
+
+As mentioned above the `Event` type was changed to be a wrapper around the
+native OS type, in turn this removed the `Ready` type in favour of having
+methods on `Event` to check for the readiness indicators and getting the
+`Token`. The `Ready` type, as used in registering, was changed to `Interests` to
+better reflect its usage. The API of `Interests` was also changed to take
+advanced of the (somewhat) new associated constants, ensuring that is no longer
+possible to register an event source with empty interests.
+
+In version 0.7 Mio registers all source with edge triggers, removing the need
+for `PollOpt`. How to use edge triggers is explained below.
+
+The trait that defined how to register an event source in version 0.6 was called
+`Evented`, this was changed to `Source` and is now referenced to as
+`event::Source`, as the type lives inside the `event` module. The same three
+methods live on the trait but are changed as described above.
+
+Finally the register functions have moved from `Poll` to a new `Registry` type
```

Comment:
Are the functions all literally named "register"?  If so us backticks.  But if they have different names and are related merely through the general theme of registration, then s/register/registration/